core/src/main/scala/org/apache/spark/storage/DiskStore.scala

```diff
@@ -208,7 +209,7 @@ private class EncryptedBlockData(
     conf: SparkConf,
     key: Array[Byte]) extends BlockData {
 
-  override def toInputStream(): InputStream = Channels.newInputStream(open())
+  override def toInputStream(): InputStream = new NioBufferedFileInputStream(file)
```

Comment:
This should create a `CryptoInputStream` right? Why could we simply new a `NioBufferedFileInputStream` instead? Am I missing something?

Reply:
Sorry for the mistake, updated.